tools/shell/CMakeLists.txt

```diff
@@ -1,20 +1,25 @@
+cmake_minimum_required(VERSION 3.17)
```

Comment:
Do we need this minimum version? It seems to break several CI runs.

Reply:
Those shell changes were added by accident after a huge merge I had to make in my fork (after that PR with cast stuff).